closed/adds/jdk/src/windows/native/jdk/crypto/jniprovider/NativeCrypto_md.c

```diff
@@ -23,27 +23,53 @@
  */
 
 #include <windows.h>
+#include <stdio.h>
+#include <jni.h>
+#include <sys/stat.h>
 
 #include "NativeCrypto_md.h"
 
+#define JAVA_DLL "java.dll"
+
+/* Prototypes. */
+jboolean GetJREPath(char *path, jint pathsize);
+jboolean GetApplicationHome(char *buf, jint bufsize);
+int JLI_Snprintf(char* buffer, size_t size, const char* format, ...);
+
 /* Load the crypto library (return NULL on error) */
 void * load_crypto_library() {
     void * result = NULL;
     const char *libname;
     const char *oldname = "libeay32.dll";
+    char opensslpath[MAX_PATH];
+    char libpathname[MAX_PATH];
+    int rc;
```

Comment:
This should be declared in the smallest block where it's used.

Reply:
Done